### PR TITLE
Fix config-daemon not waiting for drain to complete

### PR DESCRIPTION
**Pull request by SchSeba**

### CODE REVIEW

File: pkg/daemon/daemon.go [hunk 1]
```diff
@@ -484,10 +484,14 @@ func (dn *Daemon) nodeStateSyncHandler() error {
 	if reqDrain || !utils.ObjectHasAnnotation(dn.desiredNodeState,
 		consts.NodeStateDrainAnnotationCurrent,
 		consts.DrainIdle) {
-		if err := dn.handleDrain(reqReboot); err != nil {
+		shouldReconcile, err := dn.handleDrain(reqReboot)
```
Comment (ykulazhenkov) [inline pkg/daemon/daemon.go:487]: can you rename the variable to something like `inProgress` or `drainInProgress`?  For me `shouldReconcile` looks more like "should continue the current reconcile cycle", but not like "retry on next reconcile cycle" (the meaning which we want, If I understood correctly)

Comment (SchSeba) [reply pkg/daemon/daemon.go]: sure! done

File: pkg/daemon/daemon.go [hunk 2]
```diff
@@ -484,10 +484,14 @@ func (dn *Daemon) nodeStateSyncHandler() error {
...
 		consts.DrainIdle) {
-		if err := dn.handleDrain(reqReboot); err != nil {
+		drainInProcess, err := dn.handleDrain(reqReboot)
+		if err != nil {
 			log.Log.Error(err, "failed to handle drain")
 			return err
 		}
+		if drainInProcess {
+			return nil
+		}
```
[the remark this answers is not in the document]
Comment (SchSeba) [reply pkg/daemon/daemon.go:494]: so if we req drain but the operator complete the drain we are good

https://github.com/k8snetworkplumbingwg/sriov-network-operator/pull/682/files/07254b5d6032f2650affb88d5c4f2a0e443a427e#diff-a53b7b593d3d778e62eaeeafa40088656f9212bfa2c2b7991df15fa78e60b0f0R573

```
// done with the drain we can continue with the configuration
	if utils.ObjectHasAnnotation(dn.desiredNodeState, consts.NodeStateDrainAnnotationCurrent, consts.DrainComplete) {
		log.Log.Info("handleDrain(): the node complete the draining")
		return false, nil
	}
```

[the remark this answers is not in the document]
Comment (zeeke) [reply pkg/daemon/daemon.go:494]: Ok, than we assume reqDrain is always true during the draining process. correct?

**Review by zeeke — APPROVED** [review]
LGTM

File: pkg/daemon/daemon.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (SchSeba) [reply pkg/daemon/daemon.go:494]: I just checked the code and that is the case for all the plugins

**Pull request merged**